### PR TITLE
Remove `_NATIVE_AMP_AVAILABLE` checks

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

PyTorch 1.6 is our minimum supported version which does include native AMP. The check is not necessary anymore.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]

**Comment by daniellepintz** [comment]
@carmocca it seems this PR caused `test_cpu_amp_precision_context_manager` to start failing in PT 1.10

It wasn't failing in the previous PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/9748/checks?check_run_id=3740129335

File: tests/plugins/test_amp_plugins.py [hunk 1]
```diff
@@ -201,10 +201,6 @@ def test_cpu_amp_precision_throws_error(tmpdir):
 
 
 @pytest.mark.skipif(not _TORCH_CPU_AMP_AVAILABLE, reason="Torch CPU AMP is not available.")
-@RunIf(
-    min_gpus=1,
```
Comment (daniellepintz) [inline tests/plugins/test_amp_plugins.py:205]: why did you remove `min_gpus=1,` here?

Comment (awaelchli) [reply tests/plugins/test_amp_plugins.py:205]: @daniellepintz I suppose because 
given the setting `(precision="bf16", use_cpu=True)` and the check assert `isinstance(context_manager, torch.cpu.amp.autocast)` should pass regardless of whether we are on a cpu-only machien or one with gpus.

Comment (daniellepintz) [reply tests/plugins/test_amp_plugins.py:205]: makes sense, thanks!

**Comment by daniellepintz** [comment]
nevermind, this PR isn't the culprit, #9211 was the culprit but this PR just exposed the test to running be removing the `@RunIf(min_gpus=1)` check. Fix in #9809
